### PR TITLE
fix(assistant): Render tool-call parts in chat thread

### PR DESCRIPTION
**Pull request by SoumyaRaikwar**
## Which problem is this PR solving?
Part of #3932 
- Tool-call message parts from the AG-UI runtime are silently dropped in the Ask Jaeger chat panel. When the AI agent calls an MCP tool (e.g. `get_services`, `read_skill`), the user sees nothing while the tool runs  and no confirmation it completed, only the final text reply appears.

## Description of the changes
- Add `JaegerToolCallIndicator` component that renders tool-call parts  inline in the chat thread:
  - **While running** (`status.type === 'running'` or `requires-action`): shows *"Calling tool_name…"* in italic
  - **When complete**: shows **"Called tool_name"** as a collapsible `<details>` block with Input/Output sections (collapsed by default)
  - **On failure** (`status.type === 'incomplete'` or `isError`): shows **"Failed tool_name"** with an error-colored border
- Reuse `react-json-view-lite` (`JsonView`) from `AttributesTable` for JSON results; plain text results fall back to a `<pre>` block
- Extend the `MessagePrimitive.Parts` render function in  `JaegerThreadMessageBody` to handle `part.type === 'tool-call'` alongside the existing `'text'` branch
- Use `part.status` for lifecycle state instead of inferring from `result === undefined`
- Conditionally render the Output section only when `result` is defined

## Screenshots
Screenshot from 2026-06-29 02-39-31

## Screencast
[Screencast from 2026-06-29 02-38-12.webm](https://github.com/user-attachments/assets/93b769e5-586f-4a56-975f-5bc1bb20a75a)

## How was this change tested?
- Added 4 new test cases to `JaegerAssistantPanel.test.jsx`:
  - Running indicator renders "Calling tool_name…" when result is undefined
  - Completed indicator renders "Called tool_name" with result text
  - Long results are truncated at 300 chars with ellipsis
  - Text parts still render correctly alongside tool-call parts (regression)
- Verified end-to-end with Jaeger backend + Gemini AI sidecar: tool calls for `get_services` and `read_skill` render correctly in the chat panel
- All 3217 tests pass, lint and build clean

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully: `make lint test`

## AI Usage in this PR (choose one)
See [AI Usage Policy](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#ai-usage-policy).
- [ ] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [x] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

### CODE REVIEW

**Comment by SoumyaRaikwar** [comment]
@jkowall  tool calls currently show raw JSON. So updated the Ask Jaeger panel to show tool calls as expandable blocks (description in header, raw output inside). i think its much more cleaner now 

[Screencast from 2026-06-29 02-38-12.webm](https://github.com/user-attachments/assets/344f1511-9a0a-44f2-ad03-e8b5d75389a0)

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 1]
```diff
@@ -31,21 +31,79 @@ function JaegerAssistantBootstrap() {
...
+function JaegerToolCallIndicator({
+  toolName,
+  argsText,
+  result,
+}: {
+  toolName: string;
+  argsText?: string;
+  result?: unknown;
+}) {
+  const isRunning = result === undefined;
```
Comment (jkowall) [inline packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: This should key off the part lifecycle status from `MessagePrimitive.Parts`, not the presence of `result`. A tool call can complete without returning a result payload, in which case assistant-ui marks the part complete while `result` remains `undefined`; this code would keep rendering `Calling tool…` forever. Please pass `part.status` into `JaegerToolCallIndicator` and treat only `running` or `requires-action` as the in-progress states. A regression test with `{ type: 'tool-call', status: { type: 'complete' }, result: undefined }` would cover the edge case.

**Review by jkowall — CHANGES_REQUESTED** [review]
Thanks for tightening up the tool-call display. One lifecycle edge case needs fixing before merge: the renderer should use assistant-ui's `part.status` instead of inferring running state from `result === undefined`.

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx (hunk 1 above)
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: Updated the indicator to key off `part.status.type` instead of `result === undefined`:  only `running` and `requires-action` now show the "Calling…" state. Added a regression test with `{ type: 'tool-call', status: { type: 'complete' }, result: undefined }` to cover the edge case you described.

Also switched the completed state to a collapsible `<details>` element with Input/Output sections while at it.

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 2]
```diff
@@ -31,21 +31,82 @@ function JaegerAssistantBootstrap() {
...
 
+function formatToolResult(result: unknown): string {
+  if (typeof result === 'string') return result;
+  try {
+    const json = JSON.stringify(result);
+    return json ?? String(result);
+  } catch {
+    return String(result);
+  }
+}
```
[the remark this answers is not in the document]
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:53]: Skipping this suggestion:  `JSON.stringify` never returns `null` for non-null inputs (it either returns a string or throws), so the `json ?? String(result)` fallback is dead code. Also, we intentionally removed truncation since the result is now inside a collapsible `<details>` element where the full content is appropriate. The current implementation is simpler and handles the same cases.

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 3]
```diff
@@ -31,21 +31,84 @@ function JaegerAssistantBootstrap() {
...
-              component="div"
-              className="JaegerAssistantPanel-messageText"
-              smooth={false}
-            />
-          ) : null
-        }
+        {({ part }) => {
+          if (part.type === 'text') {
+            return (
+              <MessagePartPrimitive.Text
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:129]: How come this component receives no input from the actual agent response? What exactly does it display?

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 4]
```diff
@@ -31,21 +31,84 @@ function JaegerAssistantBootstrap() {
...
+  argsText,
+  result,
+  status,
+}: {
+  toolName: string;
+  argsText?: string;
+  result?: unknown;
+  status?: { type: string };
+}) {
+  const isRunning = status?.type === 'running' || status?.type === 'requires-action';
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:83]: what about other states like incomplete or error, `isError`?
```
const isDone = status?.type === 'complete' || status?.type === 'incomplete';
const isFailed = status?.type === 'incomplete' || part.isError;
```

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 5]
```diff
@@ -31,21 +31,84 @@ function JaegerAssistantBootstrap() {
   return null;
 }
 
+function formatToolResult(result: unknown): string {
+  if (typeof result === 'string') return result;
+  try {
+    const json = JSON.stringify(result);
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: We do have a widget that can display JSON nicely, it's used in the Span Details for JSON encoded attributes. Can we reuse it here?

**Review by yurishkuro — COMMENTED** [review]
this is very nice!

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx (hunk 3 above)
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:129]: `MessagePartPrimitive.Text ` is a context-driven component from `@assistant-ui/react `.  it subscribes to the runtime store and re-renders as streaming text deltas arrive via SSE. The `MessagePrimitive.Parts` wrapper iterates over message parts and sets up context for each one; the primitive reads the text content from that context, not via explicit props. We could render part.text directly (like we do with `part.toolName` for tool calls), but the primitive handles live streaming updates without us managing store subscriptions manually.

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx (hunk 4 above)
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:83]: Updated  the indicator now tracks incomplete and isError states. Failed tool calls render "Failed <tool_name>" with an error-colored border. Added tests for both isError: true and status: { type: 'incomplete' } cases.

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx (hunk 5 above)
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: done

**Comment by SoumyaRaikwar** [comment]
[Screencast from 2026-06-30 00-27-03.webm](https://github.com/user-attachments/assets/0015ca1e-a94f-4368-b92b-3af938177caa)

updated

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 6]
```diff
@@ -31,21 +32,136 @@ function JaegerAssistantBootstrap() {
...
+          </div>
+        )}
+        {result !== undefined && (
+          <div className="JaegerAssistantPanel-toolCallSection">
+            <span className="JaegerAssistantPanel-toolCallLabel">Output</span>
+            <ToolCallResultValue result={result} />
+          </div>
+        )}
+      </div>
+    </details>
```
[the remark this answers is not in the document]
Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx:116]: updated the pr description

File: packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx [hunk 7]
```diff
@@ -31,21 +32,136 @@ function JaegerAssistantBootstrap() {
...
+function formatToolResult(result: unknown): string {
+  if (typeof result === 'string') return result;
+  try {
+    return JSON.stringify(result);
+  } catch {
+    return String(result);
+  }
+}
+
+const jsonViewStyles = {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: The style bridging is already in AttributesTable.tsx (lines 82–97) — the exact same 12-property style object inlined into <JsonView>. The PR copies it verbatim into jsonViewStyles instead of extracting it into a shared constant (e.g. in a utils/jsonView.ts or in AttributesTable.tsx itself as an export). This is straightforward duplication that should be factored out.

Comment (SoumyaRaikwar) [reply packages/jaeger-ui/src/components/App/JaegerAssistantPanel.tsx]: Extracted the json-markup style object into a shared constant at `src/utils/jsonViewStyles.ts`. Both `AttributesTable` and `JaegerAssistantPanel` now import from there:  no more duplication.

**Review by yurishkuro — APPROVED** [review]
